Linked List/reverse_linked_list.cpp

```diff
@@ -0,0 +1,31 @@
+//iterative solution-----
```

Comment:
Could you add explanation, description and sample io to this solution

Reply:
okay sure